src/simulator/ackermann_model.cpp

```diff
@@ -0,0 +1,99 @@
+#include "simulator/ackermann_model.h"
+#include "shared/util/timer.h"
+#include "shared/math/math_util.h"
+
+using f1tenth_simulator::AckermannCurvatureDriveMsg;
+using math_util::AngleDiff;
+using math_util::AngleMod;
+using std::string;
+using std::isfinite;
+
+namespace ackermann {
+
+CONFIG_FLOAT(cMinTurnR, "min_turn_radius");
+CONFIG_FLOAT(cMaxAccel, "max_accel");
+CONFIG_FLOAT(cMaxSpeed, "max_speed");
+CONFIG_FLOAT(cAngularErrorBias, "angular_error_bias");
+CONFIG_FLOAT(cAngularErrorRate, "angular_error_rate");
+CONFIG_STRING(cDriveTopic, "drive_callback_topic");
+
+
+AckermannModel::AckermannModel(const string& config_file, ros::NodeHandle n) :
```

Comment:
This copies the ros NodeHandle. Unclear what the implications of that are. Why not take in a pointer?